_targets.R

```diff
@@ -165,13 +157,73 @@ targets_list <- list(
   ##### Munge GDP output into NetCDF files that will feed into GLM #####
 
   # Munge GCM variables into useable GLM variables and correct units
+  # For each tile-gcm combo this function returns a list with 2 elements:
+  # 1) file_out - the name of the munged output file, which contains data
+  # only for queried cells in that tile that returned data for all variables, and
+  # 2) cell_info - a tibble with a row for each queried cell in that tile, indicating
+  # whether or not that cell is missing data for *any* variable for that gcm
+  tar_target(
+    glm_ready_gcm_data_list,
+    munge_notaro_to_glm(gcm_data_raw_feather, gcm_names, unique(query_cells_centroids_list_by_tile$tile_no)),
+    pattern = map(gcm_data_raw_feather, cross(query_cells_centroids_list_by_tile, gcm_names, gcm_dates_df))
+  ),
+
+  # Combine the `file_out` elements of the gcm_ready_gcm_data_list branches
+  # into a single vector of file targets
   tar_target(
     glm_ready_gcm_data_feather,
-    munge_notaro_to_glm(gcm_data_raw_feather),
-    pattern = map(gcm_data_raw_feather),
+    glm_ready_gcm_data_list$file_out,
+    pattern = map(glm_ready_gcm_data_list),
```

Comment:
So interestingly, `targets` doesn't actually return a nested list, which is what you've got laid out above and what I'd expected, too. For example, for the 24 tile-gcm combos we're working with here, `glm_ready_gcm_data_list` is not a list of 24 elements, with nested sub lists, but a list of 48 elements.:
![image](https://user-images.githubusercontent.com/54007288/155412227-73a29a51-ba09-42dc-8a4c-400d7b54150c.png)
Therefore if I try to use `purrr::modify_depth()` as you have here, I get a 'subscript out of bounds' error.

Reply:
And if I use `purrr::modify_depth(0, "file_out")` or just `glm_ready_gcm_data_list$file_out` it errors because it doesn't return a character vector.

Reply:
Aha! BUT if I nest the list within a list [here](https://github.com/hcorson-dosch/lake-temperature-model-prep/blob/gcm_address_na_cells/7_drivers_munge/src/GCM_driver_utils.R#L465) then your code does work, so I'll make that change

Reply:
The other alternative is the use the names of the returned list to your advantage rather than nesting further to use that unnesting function.

```r
# This would return all of the files out
unlist(gcm_ready_gcm_data_list[grepl("file_out", names(gcm_ready_gcm_data_list))])

# This would return all of the dfs
bind_rows(gcm_ready_gcm_data_list[grepl("cell_info", names(gcm_ready_gcm_data_list))])
```

Reply:
I think I like the approach I switched to, since it returns a list of the expected length for `glm_ready_gcm_data_list`

Reply:
Actually I'm going to switch to your latest approach, since the double nesting is a bit odd to do just to use that unnesting function, as you noted

Reply:
Also, out of curiosity, why are you against using mapping in this case? Because it's not actually necessary to map over the upstream target in order to build the desired downstream target, and we should only use mapping where necessary? Or because of the extra console statements? Just wondering, b/c the branched approach did work, and changing it meant further changes to downstream targets (e.g., `gcm_data_daily_feather_group_by_gcm`). I guess to me, the grepl approach certainly works, but it feels like using a workaround rather than a built-in `targets` capability. Even if the mapping in this case would just be for functional reasons, rather than a necessary coding step to build the desired downstream target.

Reply:
Also - I know you have a lot on your plate, so if you don't have time for a discussion no worries -- the revised code works well.

Reply:
Hmm I guess I am not seeing it as a workaround, more of an alternative to branching. I think of branching as a tool to use when you can't do a vectorized operation or it is expensive to run the target and we want to be choosy about what causes the rebuilds (maybe this is true for the cell_info extraction here, but since they are being combined into one df, the full target would build every time there was a change anyways). I think we want to avoid branching when possible because `targets` saves files for each branch and target that builds, which creates additional overhead and run time to the full pipeline. I am thinking of it kind of as the pipeline version of doing a vectorized operation vs building a `for` loop to do it. For example,

```r
vec <- 1:6

# If you wanted to add 1 to the vector, you would probably just write
new_vec <- vec + 1 

# You probably wouldn't write bc that's a lot more code and overhead for the simpler action
new_vec <- c()
for(i in seq_along(vec)) {
    new_vec <- c(new_vec, vec[i] + 1)
}
```

Reply:
> `targets` saves files for each branch and target that builds, which creates additional overhead and run time to the full pipeline. I am thinking of it kind of as the pipeline version of doing a vectorized operation vs building a `for` loop to do it.

Thanks, this is helpful context! I didn't realize that branching also causes unnecessary I/O under the surface. And that analogy of vectorized operation vs. `for` loop is really useful.